### ISSUE TITLE
Upgrade to 6.0 gives an 'Access to path' error 

### ISSUE THREAD

**Issue by ElimAdmin** [SparkDevNetwork/Rock#1935]
I have just upgraded from 5.4 to 5.5. successfully.  
The upgrade page then gave notification that I needed .NET 4.5.2. to upgrade to v6.0.
I installed .NET 4.5.2 on the server.
Now, on loading the Rock Update page, I receive the error message:

Whoa that wasn't supposed to happen
Here's what happened... ...actually, I'm not sure what happened here: An error occurred while loading packages from 'http://update.rockrms.com/F/rock/api/v2/': Access to the path 'C:\Windows\system32\config\systemprofile\AppData\Roaming\NuGet' is denied. 

I noticed from the Diagnostic output (see below), that the 
Execution Location:: C:\Windows\Microsoft.NET\Framework\v4.0.30319\Temporary ASP.NET Files\root\7c5b939b\e1ab6f53\App_Web_dlopo2jd.dll

seems to still reference .NET framework 4.0.

Is this a Rock issue, or do I need help on the server side of things to make sure that .NET 4.5.2 is loading properly.  Using Plesk to view the settings on the server (3essentials), the ASP.NET framework is showing as 4.5.2.  I have rebooted the server.  Any advice appreciated.  Ken

DIAGNOSTIC FILE:

Version:: Rock McKinley 5.5
Database:: Name: elimrock 
 Server: localhost
Database Version: Microsoft SQL Server 2012 - 11.0.2218.0 (X64) 
	Jun 12 2012 13:05:25 
	Copyright (c) Microsoft Corporation
	Express Edition (64-bit) on Windows NT 6.2 <X64> (Build 9200: ) (Hypervisor)

Database Size: Unlimited
Allow Snapshot Isolation: No
Is Read Committed Snapshot On: No

Execution Location:: C:\Windows\Microsoft.NET\Framework\v4.0.30319\Temporary ASP.NET Files\root\7c5b939b\e1ab6f53\App_Web_dlopo2jd.dll
Migrations:: Last Core Migration: 201606061308109_ClearPreviousMigrations5
<table class='table table-condensed'><tr><th>Plugin Assembly</th><th>Migration Name</th><th>Number</th></tr><tr><td>Rock</td><td>FixGroupAndBenevolenceSecurity</td><td>7</td></tr><tr><td>Rock.Migrations</td><td>StatementProcs</td><td>6</td></tr></table>
Cache:: <p><strong>Cache Items:</strong>
 6849</p>
<p><strong>Cache Memory Limit:</strong>
 1,887,436,800 (bytes)</p>
<p><strong>Physical Memory Limit:</strong>
 98 %</p>
<p><strong>Polling Interval:</strong>
 00:02:00</p>

Routes:: 
{resource}.axd/{*pathInfo}
AddTransaction
api/{*odataPath}
api/{controller}
api/{controller}({key})
api/{controller}/{id}
api/{controller}/AttributeValue/{id}
api/{controller}/DataView/{id}
api/{controller}/LaunchWorkflow/{id}
api/{controller}/SetContext/{id}
api/taggeditems/{entityTypeId}/{ownerid}/{entityguid}/{entityqualifier}/{entityqualifiervalue}
api/tags/{entityTypeId}/{ownerid}/{entityqualifier}/{entityqualifiervalue}
api/tags/availablenames/{entityTypeId}/{ownerid}/{name}/{entityguid}/{entityqualifier}/{entityqualifiervalue}
BlockProperties/{BlockId}
BulkUpdate/{Set}
Business/Search/{SearchType}
ChangePassword
checkin
checkin/{KioskId}/{CheckinConfigId}/{GroupTypeIds}
checkin/{KioskId}/{GroupTypeIds}
checkin/ability
checkin/family
checkin/group
checkin/grouptype
checkin/location
checkin/people
checkin/person
checkin/scheduledlocations
checkin/search
checkin/success
checkin/time
checkin/welcome
CheckinManager
ckeditorplugins/RockFileBrowser
ckeditorplugins/RockMergeField
Communication
Communication/{CommunicationId}
ConfirmAccount
ContactUs
DISC
DISC/{rckipid}
EditFamily/{PersonId}/{GroupId}
EventItem/{EventItemId}
Events
EY
FollowingSuggestionList
ForgotUserName
Group/{groupId}
Group/Search/{SearchType}
GroupMember/{GroupMemberId}
GroupType/{GroupTypeId}
kiosk
LaunchWorkflow/{WorkflowTypeId}
MergeTemplate/{Set}
MyAccount
MyDashboard
MyDashboardWorkflow/{WorkflowId}
MyDashboardWorkflowEntry/{WorkflowTypeId}
MyDashboardWorkflowEntry/{WorkflowTypeId}/{WorkflowId}
MyWorkflow/{WorkflowId}
MyWorkflowEntry/{WorkflowTypeId}
MyWorkflowEntry/{WorkflowTypeId}/{WorkflowId}
NewAccount
NewFamily
Package/{PackageId}
page/{PageId}
PageProperties/{Page}
Pages/{EditPage}
Person/{PersonId}
Person/{PersonId}/Benevolence
Person/{PersonId}/Contributions
Person/{PersonId}/Edit
Person/{PersonId}/ExtendedAttributes
Person/{PersonId}/Groups
Person/{PersonId}/History
Person/{PersonId}/Security
Person/Search/{SearchType}
PersonDuplicate/{PersonId}
PersonMerge/{Set}
PhotoRequest/OptOut/{Person}
PhotoRequest/Upload/{rckipid}
Prayer
Registration
Registration/{CampusId}/{GroupId}
Registration/{Slug}
RegistrationInstance/{RegistrationInstanceId}
resources
RockShop
RockShop/Account
RockShop/Purchases
Schedules
Secure/{EntityTypeId}/{EntityId}
SecurityRoles
Site/{Action}
Site/{Action}/{SiteId}
Sites
SystemEmailCategories
SystemInfo
Transaction/{transactionId}
unstop
Unstop15Rego
unstoppable
unstp15
Unsubscribe/{Person}
Workflow/{WorkflowId}
WorkflowEntry/{WorkflowTypeId}
WorkflowEntry/{WorkflowTypeId}/{WorkflowId}
Worklow
youth
ZoneBlocks/{EditPage}/{ZoneName}

Server Variables:: 
ALL_HTTP: HTTP_CONNECTION:keep-alive
HTTP_CONTENT_LENGTH:8001
HTTP_CONTENT_TYPE:application/x-www-form-urlencoded
HTTP_ACCEPT:text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
HTTP_ACCEPT_ENCODING:gzip, deflate
HTTP_ACCEPT_LANGUAGE:en-US,en;q=0.5
HTTP_COOKIE:
HTTP_HOST:porirua.elim.org.nz
HTTP_REFERER:http://porirua.elim.org.nz/SystemInfo?t=System Information&pb=&sb=Done
HTTP_USER_AGENT:Mozilla/5.0 (Windows NT 6.1; rv:50.0) Gecko/20100101 Firefox/50.0
HTTP_UPGRADE_INSECURE_REQUESTS:1

ALL_RAW: Connection: keep-alive
Content-Length: 8001
Content-Type: application/x-www-form-urlencoded
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.5
Host: porirua.elim.org.nz
Referer: http://porirua.elim.org.nz/SystemInfo?t=System Information&pb=&sb=Done
User-Agent: Mozilla/5.0 (Windows NT 6.1; rv:50.0) Gecko/20100101 Firefox/50.0
Upgrade-Insecure-Requests: 1

APPL_MD_PATH: /LM/W3SVC/6/ROOT
APPL_PHYSICAL_PATH: C:\Inetpub\vhosts\porirua.elim.org.nz\httpdocs\
AUTH_TYPE: Forms
AUTH_USER: ken
AUTH_PASSWORD: 
LOGON_USER: ken
REMOTE_USER: ken
CERT_COOKIE: 
CERT_FLAGS: 
CERT_ISSUER: 
CERT_KEYSIZE: 
CERT_SECRETKEYSIZE: 
CERT_SERIALNUMBER: 
CERT_SERVER_ISSUER: 
CERT_SERVER_SUBJECT: 
CERT_SUBJECT: 
CONTENT_LENGTH: 8001
CONTENT_TYPE: application/x-www-form-urlencoded
GATEWAY_INTERFACE: CGI/1.1
HTTPS: off
HTTPS_KEYSIZE: 
HTTPS_SECRETKEYSIZE: 
HTTPS_SERVER_ISSUER: 
HTTPS_SERVER_SUBJECT: 
INSTANCE_ID: 6
INSTANCE_META_PATH: /LM/W3SVC/6
LOCAL_ADDR: 216.85.168.90
PATH_INFO: /SystemInfo
PATH_TRANSLATED: C:\Inetpub\vhosts\porirua.elim.org.nz\httpdocs\SystemInfo
QUERY_STRING: t=System+Information&pb=&sb=Done
REMOTE_ADDR: 203.109.201.165
REMOTE_HOST: 203.109.201.165
REMOTE_PORT: 59313
REQUEST_METHOD: POST
SCRIPT_NAME: /SystemInfo
SERVER_NAME: porirua.elim.org.nz
SERVER_PORT: 80
SERVER_PORT_SECURE: 0
SERVER_PROTOCOL: HTTP/1.1
SERVER_SOFTWARE: Microsoft-IIS/8.5
URL: /SystemInfo
HTTP_CONNECTION: keep-alive
HTTP_CONTENT_LENGTH: 8001
HTTP_CONTENT_TYPE: application/x-www-form-urlencoded
HTTP_ACCEPT: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
HTTP_ACCEPT_ENCODING: gzip, deflate
HTTP_ACCEPT_LANGUAGE: en-US,en;q=0.5
HTTP_HOST: porirua.elim.org.nz
HTTP_REFERER: http://porirua.elim.org.nz/SystemInfo?t=System Information&pb=&sb=Done
HTTP_USER_AGENT: Mozilla/5.0 (Windows NT 6.1; rv:50.0) Gecko/20100101 Firefox/50.0
HTTP_UPGRADE_INSECURE_REQUESTS: 1

**Comment by arranf** [comment]
`'C:\Windows\system32\config\systemprofile\AppData\Roaming\NuGet' is denied.` is the key line.
It looks like the IIS process doesn't have access to `AppData`, you'll want to change your permissions on that folder.

**Comment by ElimAdmin** [comment]
Thanks @arranf.  This folder has full permissions for SYSTEM and Administrators.  What should the permissions be?  Or what other user needs permissions?

**Comment by nairdo** [comment]
The reference to "C:\Windows\Microsoft.NET\Framework\v4.0.30319" is normal. In general, ASP.Net 4.5.2 still reports as .Net v4. Looking at our own site (rockrms.com) on LiquidWeb here are a few things I noticed:

1. We don't have a `NuGet` folder under `C:\Windows\system32\config\systemprofile\AppData\Roaming`
2. Our app pool settings are:
   1.  "Enable 32-Bit Applications" set to **False**. (I'm not sure if this is relevant, but I thought I'd document it here.)
   2. "Managed Pipeline Mode" set to "Integrated"
   3. "Identity" set to "ApplicationPoolIdentity" which is a little different than what is shown in this screenshot: http://www.rockrms.com/Rock/LetsFixThis#IncorrectDotNETVersion
 
I'm mostly wondering if there is a different version of NuGet core in the [GAC](https://en.wikipedia.org/wiki/Global_Assembly_Cache) that's the causing this issue [also reported with someone's LiquidWeb](https://www.rockrms.com/Rock/Ask/Using/Questions/1316) system. I'm also wondering if that NuGet folder was gone, if NuGet would behave differently.

**Comment by nairdo** [comment]
@ElimAdmin any comments on this?

**Comment by ElimAdmin** [comment]
Hi @nairdo .  Happy New Year! 
I have deleted the NuGet directory.  The app pool settings are the same as those you've listed.  When I go into Rock Update I still get the same error message.  Do I need to roll back and retry the install?  Or do I need to try reinstalling .NET on the server?

**Comment by ElimAdmin** [comment]
@nairdo I have restored to what I thought was my backup prior to the attempted v6 upgrade, but I still get the same error message in the Rock Update page, suggesting this is a .NET issue?   Not sure where to go from here.

Here's the current diagnostic file:
Version:: Rock McKinley 5.5
Database:: Name: elimrock 
 Server: localhost
Database Version: Microsoft SQL Server 2012 - 11.0.2218.0 (X64) 
	Jun 12 2012 13:05:25 
	Copyright (c) Microsoft Corporation
	Express Edition (64-bit) on Windows NT 6.2 <X64> (Build 9200: ) (Hypervisor)

Database Size: Unlimited
Allow Snapshot Isolation: No
Is Read Committed Snapshot On: No

Execution Location:: C:\Windows\Microsoft.NET\Framework\v4.0.30319\Temporary ASP.NET Files\root\7c5b939b\e1ab6f53\App_Web_dlopo2jd.dll
Migrations:: Last Core Migration: 201606061308109_ClearPreviousMigrations5
<table class='table table-condensed'><tr><th>Plugin Assembly</th><th>Migration Name</th><th>Number</th></tr><tr><td>Rock</td><td>FixGroupAndBenevolenceSecurity</td><td>7</td></tr><tr><td>Rock.Migrations</td><td>StatementProcs</td><td>6</td></tr></table>
Cache:: <p><strong>Cache Items:</strong>
 6936</p>
<p><strong>Cache Memory Limit:</strong>
 1,887,436,800 (bytes)</p>
<p><strong>Physical Memory Limit:</strong>
 98 %</p>
<p><strong>Polling Interval:</strong>
 00:02:00</p>

Routes:: 
{resource}.axd/{*pathInfo}
AddTransaction
api/{*odataPath}
api/{controller}
api/{controller}({key})
api/{controller}/{id}
api/{controller}/AttributeValue/{id}
api/{controller}/DataView/{id}
api/{controller}/LaunchWorkflow/{id}
api/{controller}/SetContext/{id}
api/taggeditems/{entityTypeId}/{ownerid}/{entityguid}/{entityqualifier}/{entityqualifiervalue}
api/tags/{entityTypeId}/{ownerid}/{entityqualifier}/{entityqualifiervalue}
api/tags/availablenames/{entityTypeId}/{ownerid}/{name}/{entityguid}/{entityqualifier}/{entityqualifiervalue}
BlockProperties/{BlockId}
BulkUpdate/{Set}
Business/Search/{SearchType}
ChangePassword
checkin
checkin/{KioskId}/{CheckinConfigId}/{GroupTypeIds}
checkin/{KioskId}/{GroupTypeIds}
checkin/ability
checkin/family
checkin/group
checkin/grouptype
checkin/location
checkin/people
checkin/person
checkin/scheduledlocations
checkin/search
checkin/success
checkin/time
checkin/welcome
CheckinManager
ckeditorplugins/RockFileBrowser
ckeditorplugins/RockMergeField
Communication
Communication/{CommunicationId}
ConfirmAccount
ContactUs
DISC
DISC/{rckipid}
EditFamily/{PersonId}/{GroupId}
EventItem/{EventItemId}
Events
EY
FollowingSuggestionList
ForgotUserName
Group/{groupId}
Group/Search/{SearchType}
GroupMember/{GroupMemberId}
GroupType/{GroupTypeId}
kiosk
LaunchWorkflow/{WorkflowTypeId}
MergeTemplate/{Set}
MyAccount
MyDashboard
MyDashboardWorkflow/{WorkflowId}
MyDashboardWorkflowEntry/{WorkflowTypeId}
MyDashboardWorkflowEntry/{WorkflowTypeId}/{WorkflowId}
MyWorkflow/{WorkflowId}
MyWorkflowEntry/{WorkflowTypeId}
MyWorkflowEntry/{WorkflowTypeId}/{WorkflowId}
NewAccount
NewFamily
Package/{PackageId}
page/{PageId}
PageProperties/{Page}
Pages/{EditPage}
Person/{PersonId}
Person/{PersonId}/Benevolence
Person/{PersonId}/Contributions
Person/{PersonId}/Edit
Person/{PersonId}/ExtendedAttributes
Person/{PersonId}/Groups
Person/{PersonId}/History
Person/{PersonId}/Security
Person/Search/{SearchType}
PersonDuplicate/{PersonId}
PersonMerge/{Set}
PhotoRequest/OptOut/{Person}
PhotoRequest/Upload/{rckipid}
Prayer
Registration
Registration/{CampusId}/{GroupId}
Registration/{Slug}
RegistrationInstance/{RegistrationInstanceId}
resources
RockShop
RockShop/Account
RockShop/Purchases
Schedules
Secure/{EntityTypeId}/{EntityId}
SecurityRoles
Site/{Action}
Site/{Action}/{SiteId}
Sites
SystemEmailCategories
SystemInfo
Transaction/{transactionId}
unstop
Unstop15Rego
unstoppable
unstp15
Unsubscribe/{Person}
Workflow/{WorkflowId}
WorkflowEntry/{WorkflowTypeId}
WorkflowEntry/{WorkflowTypeId}/{WorkflowId}
Worklow
youth
ZoneBlocks/{EditPage}/{ZoneName}

Server Variables:: 
ALL_HTTP: HTTP_CONNECTION:keep-alive
HTTP_CONTENT_LENGTH:8101
HTTP_CONTENT_TYPE:application/x-www-form-urlencoded
HTTP_ACCEPT:text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
HTTP_ACCEPT_ENCODING:gzip, deflate
HTTP_ACCEPT_LANGUAGE:en-US,en;q=0.5
HTTP_COOKIE:_ga=GA1.3.565242109.1405036848; ASP.NET_SessionId=dlbkmgqvhjxakg1vu3yqg5te; .ROCK=D1FA834FF636337949EBD0A29B2DEB25A6436BA67C3828109CCC4DB2D12C749636625F855EF844DB17DBA0855C0409272810FA6F0BB1A832F927D56BC65ADC3B54A322965C8503D63629F0878AE936F8ACB32D35
HTTP_HOST:porirua.elim.org.nz
HTTP_REFERER:http://porirua.elim.org.nz/SystemInfo?t=System%20Information&pb=&sb=Done
HTTP_USER_AGENT:Mozilla/5.0 (Windows NT 6.1; rv:50.0) Gecko/20100101 Firefox/50.0
HTTP_UPGRADE_INSECURE_REQUESTS:1

ALL_RAW: Connection: keep-alive
Content-Length: 8101
Content-Type: application/x-www-form-urlencoded
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.5
Cookie: _ga=GA1.3.565242109.1405036848; ASP.NET_SessionId=dlbkmgqvhjxakg1vu3yqg5te; .ROCK=D1FA834FF636337949EBD0A29B2DEB25A6436BA67C3828109CCC4DB2D12C749636625F855EF844DB17DBA0855C0409272810FA6F0BB1A832F927D56BC65ADC3B54A322965C8503D63629F0878AE936F8ACB32D35
Host: porirua.elim.org.nz
Referer: http://porirua.elim.org.nz/SystemInfo?t=System%20Information&pb=&sb=Done
User-Agent: Mozilla/5.0 (Windows NT 6.1; rv:50.0) Gecko/20100101 Firefox/50.0
Upgrade-Insecure-Requests: 1

APPL_MD_PATH: /LM/W3SVC/6/ROOT
APPL_PHYSICAL_PATH: C:\Inetpub\vhosts\porirua.elim.org.nz\httpdocs\
AUTH_TYPE: Forms
AUTH_USER: ken
AUTH_PASSWORD: 
LOGON_USER: ken
REMOTE_USER: ken
CERT_COOKIE: 
CERT_FLAGS: 
CERT_ISSUER: 
CERT_KEYSIZE: 
CERT_SECRETKEYSIZE: 
CERT_SERIALNUMBER: 
CERT_SERVER_ISSUER: 
CERT_SERVER_SUBJECT: 
CERT_SUBJECT: 
CONTENT_LENGTH: 8101
CONTENT_TYPE: application/x-www-form-urlencoded
GATEWAY_INTERFACE: CGI/1.1
HTTPS: off
HTTPS_KEYSIZE: 
HTTPS_SECRETKEYSIZE: 
HTTPS_SERVER_ISSUER: 
HTTPS_SERVER_SUBJECT: 
INSTANCE_ID: 6
INSTANCE_META_PATH: /LM/W3SVC/6
LOCAL_ADDR: 216.85.168.90
PATH_INFO: /SystemInfo
PATH_TRANSLATED: C:\Inetpub\vhosts\porirua.elim.org.nz\httpdocs\SystemInfo
QUERY_STRING: t=System+Information&pb=&sb=Done
REMOTE_ADDR: 203.109.201.165
REMOTE_HOST: 203.109.201.165
REMOTE_PORT: 51022
REQUEST_METHOD: POST
SCRIPT_NAME: /SystemInfo
SERVER_NAME: porirua.elim.org.nz
SERVER_PORT: 80
SERVER_PORT_SECURE: 0
SERVER_PROTOCOL: HTTP/1.1
SERVER_SOFTWARE: Microsoft-IIS/8.5
URL: /SystemInfo
HTTP_CONNECTION: keep-alive
HTTP_CONTENT_LENGTH: 8101
HTTP_CONTENT_TYPE: application/x-www-form-urlencoded
HTTP_ACCEPT: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
HTTP_ACCEPT_ENCODING: gzip, deflate
HTTP_ACCEPT_LANGUAGE: en-US,en;q=0.5
HTTP_COOKIE: _ga=GA1.3.565242109.1405036848; ASP.NET_SessionId=dlbkmgqvhjxakg1vu3yqg5te; .ROCK=D1FA834FF636337949EBD0A29B2DEB25A6436BA67C3828109CCC4DB2D12C749636625F855EF844DB17DBA0855C0409272810FA6F0BB1A832F927D56BC65ADC3B54A322965C8503D63629F0878AE936F8ACB32D35
HTTP_HOST: porirua.elim.org.nz
HTTP_REFERER: http://porirua.elim.org.nz/SystemInfo?t=System%20Information&pb=&sb=Done
HTTP_USER_AGENT: Mozilla/5.0 (Windows NT 6.1; rv:50.0) Gecko/20100101 Firefox/50.0
HTTP_UPGRADE_INSECURE_REQUESTS: 1

**Comment by ElimAdmin** [comment]
@nairdo 
Can this issue be re-opened?  I notice there's an unresolved comment in the Rock Questions, so this is not unique to me: https://www.rockrms.com/Rock/Ask/Troubleshooting/Questions/1124

Looks like this may be a NuGet issue.   Either a version issue, or a corrupt NuGet config file issue.  Am going to look at what version I'm running, and check out possible solutions here:
https://github.com/NuGet/Home/issues/2676

"I had this same problem using TFS. What I did that fixed it was clear the nuget cache in the user profile that was running the build (%LocalAppData%\NuGet\Cache). I just emptied the directory and ran the build again, then it worked."

**Comment by ElimAdmin** [comment]
I don't even have NuGet installed.  Where to from here?

**Comment by ElimAdmin** [comment]
Refer also to question https://www.rockrms.com/Rock/Ask/Troubleshooting/Questions/1451

**Comment by ElimAdmin** [comment]
Refer also to question https://www.rockrms.com/Rock/Ask/Using/Questions/1316

If you google 
An error occurred while loading packages from Access to the path 'C:\windows\system32\config\systemprofile\AppData\Roaming\NuGet' is denied

there are some suggestions that this might be a NuGet versioning issue (when you create the package???)?

I'm currently running .NET 4.6.2.

I have manually given Everyone full access on all directories from C:\windows\system32\config down, and I manually created a NuGet directory because there wasn't one, so I no longer think that access permissions on my server is the real issue.

If you would like me to debug the Upgrade code to let you know what is happening let me know, or if there is anything else I can do to help, please let me know.

This is probably not a high priority issue, _unless I need to roll back again to try to get a condition prior to this_ - but I did roll back, and am confused now as to why I'm still seeing the problem, and I can't hold off data updates.  

I have had 3essentials confirm that all the server settings are correct for running Rock.

I did recently update Plesk on my 3essentials VPS, and note that Ronnie resolved this issue by deleting Plesk and do a manual reinstall.

**Comment by ElimAdmin** [comment]
I have just had a 'chat' with 3essentials. "Uninstalling Plesk won't be a good idea. As your server is running with Plesk, all the functionalities are tied with Plesk and removing it can cause further instability. We are aware of this issue with RockRMS in Plesk servers and are reviewing the case with our senior admins. We have opened a ticket for y ou and will keep you updated on this."

**Comment by JasonWhitehorn** [comment]
As of this morning it appears that they are no further.  I had a similar response from 3Essentials:

> Removing Plesk temporarily from the server won't be a recommended option. I request you to wait for an update from us or if you do not want Plesk on your server then you may sign up for a self-managed server without Plesk. You may browse the URL http://www.3essentials.com/rockrms-hosting/ for the plans available. However, note that it can be difficult for you to manage the server without a control panel if you do not have good technical knowledge of the services running on the server.

**Comment by edmistj** [comment]
I reached out to the owner of 3Essentials, who is a great guy, and he assures me that these tickets will be updated today with a non-plesk solution. Rock does have a built in file upload so that should fill the need of getting files to the server.

Please update this issue once the problem is resolved.

**Comment by JasonWhitehorn** [comment]
Thanks, John.  I just got this from 3Essentials:

> In speaking with the lead developer at Rock and our management team here at 3essentials, we have identified the compatibility problem of running Rock RMS on a server with Plesk. Due to specific requirements needed to run Rock RMS, we are discontinuing our managed plans for Rock servers that include Plesk.

We would like to rebuild your server on an updated platform which does not include Plesk. This will be a Windows 2016 Server, MSSQL SQL 2016, and Rock pre-installed. Any current data and files on the server would be lost upon the rebuild.

There would be no change in pricing for the same specifications of your current server. We would still manage the server, components, and the operating system as before.

If you would like to move forward with this, pease confirm by replying to this request. Or if you have any additional questions, let us know and we'll be happy to help.

Best regards,
Tim H.

They were able to make some change to it even with Plesk...but that change would still likely make it unstable even in the long run.

Thanks, guys...for your diligence in this!

**Comment by edmistj** [comment]
It sounded like it was not making the AppPool Identity not be LocalSystem. @ElimAdmin  Can you determine if that will fix your issue too? It's documented here: https://www.rockrms.com/Rock/BookContent/1/90#configuringinternetinformationservicesiis

**Comment by 3etimh** [comment]
This was indeed the issue on our Windows server stack with Plesk. By default, and for security reasons, Plesk uses it's own identity for application pools. These users are labeled as 'IWDP_'.

In our experience, Plesk on the backend does not like modifications to IIS or the server to be made outside of the panel as these are very likely to be reset during control panel updates.

If the security requirement on the AppPool for Rock is required to be LocalSystem, then Plesk should not be used for management.

**Comment by edmistj** [comment]
Thanks @3eTimH for all the effort in solving this!

**Comment by JasonWhitehorn** [comment]
Thanks guys!  Tremendous!

**Comment by ElimAdmin** [comment]
Thank you @3eTimH and @edmistj for helping resolve this issue and for providing a path forward.  We really appreciated your help!

@edmistj - this issue (and #1967?) can be closed: we will proceed with the 'no Plesk server option.'  

Because this issue was identified when we tried to upgrade from v5.5, I'm assuming that the process now is: (1) rebuild the server, (2) do a fresh install of Rock, (3) restore the database from a backup (into the new MSSQL SQL 2016), then (4) proceed with the upgrades as normal.  

@edmistj - can you confirm that a fresh install starts from a version at or prior to v5.5 so that it will pick up from our current version? (I.e. a fresh install doesn't start at a version **after** v5.5. so we won't skip any important upgrades.)  Thank you. Ken

**Comment by edmistj** [comment]
@ElimAdmin A fresh install will start at v6.1.

**Comment by ElimAdmin** [comment]
@edmistj @nairdo @arranf @JasonWhitehorn 

Just a quick follow up to say we’ve been able to upgrade from 5.5. to 6.2 successfully.

3essentials provided a rebuilt server.
I restored the db and web site.
There were some configuration settings I had to get right.  There were some changes in the web.config file required because of the move to IIS7.
I had to ensure that the user had correct settings in SQL Server in order to enable log in.
I will document these and add them later for future reference.

Thanks again for your help.
Ken
